### DISCUSSION TITLE
Are routes not loaded when executing a command? e.g. Drive.getUrl(...)

### DISCUSSION THREAD

**Discussion by mattkenefick** [adonisjs/core#3880] [Help] [answered]
I'm executing a command with `loadApp: true` and it ultimately runs a `Drive.getUrl(...)` method which attempts to do a route lookup.

It cannot find the `drive.local.serve` route which is automatically set by the framework when `serveFiles` is true. Is it possible there are no routes loaded to search through when executing as a command?

Did I forget to config something?

**Comment by thetutlage** [comment, marked as answer]
You need to commit routes to the store before using them inside a command. The following as the first statement in the command `run` method should do the job.

```ts
this.application.container.use('Adonis/Core/Route').commit()

// Now use drive
```

**Reply by mattkenefick** [reply]
Is there additional context somewhere / in documentation on why this works or how I might use it for other things?
